packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -351,21 +370,67 @@ export async function runHttpQuery(
         }),
       };
     }
-  }) as Array<Promise<ExecutionResult>>;
+  }) as Array<Promise<ExecutionResult & { extensions?: Record<string, any> }>>;
 
   const responses = await Promise.all(requests);
 
+  const responseInit: ApolloServerHttpResponse = {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  //enabling cacheControl means that the user would like the cache-control
+  //extensions we are running the proxy, so we should not strip out the cache
+  //control extension.
+  if (!optionsObject.cacheControl) {
+    responseInit.headers = {
+      ...responseInit.headers,
+      ...calculateCacheControlHeaders(responses),
+    };
+
+    //remove cacheControl headers. This could be done in production only,
```

Comment:
It's extensions, not headers (so fix the comment).

However this all seems to be weirdly intricate.  We create the CacheControlExtension object and ask it to put things in the JSON extension format, but then we turn it into headers and delete the extension. Why not just put header calculation (or perhaps, "max age and privacy tuple" calculation) into CacheControlExtension itself, and (via an option to the CacheControlExtension constructor that can be overridden) make `format` into a no-op?